From 80df3fa126a2d60a564f77b81c7d6b02063547de Mon Sep 17 00:00:00 2001
From: Gilad Ben-Yossef <gilad@benyossef.com>
Date: Sun, 26 Mar 2017 10:12:49 +0300
Subject: [PATCH] crypto: remove limit of hashstate size

The hash state size was limited to PAGE_SIZE / 8 for no reasons obvious
to the author. This does not work with some HW based crypto implementation
that may have a lot of state.

Signed-Off-By: Gilad Ben-Yossef <gilad@benyossef.com>
---
 crypto/ahash.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/crypto/ahash.c b/crypto/ahash.c
index 2ce8bcb..03e7730 100644
--- a/crypto/ahash.c
+++ b/crypto/ahash.c
@@ -536,7 +536,6 @@ static int ahash_prepare_alg(struct ahash_alg *alg)
 	struct crypto_alg *base = &alg->halg.base;
 
 	if (alg->halg.digestsize > PAGE_SIZE / 8 ||
-	    alg->halg.statesize > PAGE_SIZE / 8 ||
 	    alg->halg.statesize == 0)
 		return -EINVAL;
 
-- 
2.1.4

